specification/sections/01-flag-evaluation.md

```diff
@@ -381,6 +381,18 @@ This `flag metadata` field is intended as a mechanism for providers to surface a
 
 > Condition: `Flag metadata` **MUST** be immutable.
 
+#### Requirement 1.4.16
+
+> Methods, functions, or operations on the client **SHOULD NOT** write log messages.
```

Comment:
This is one of the few times I've thought SHOULD NOT was the best rfc-2119 keyword: https://datatracker.ietf.org/doc/html/rfc2119#section-4

I think there may be exceptions for some languages (possibly debug level logging which is not generally used unless doing some kind of issue investigation or during local development).